### PR TITLE
Require openspecfun

### PR DESCRIPTION
**Pull request by jakirkham**
Fixes https://github.com/conda-forge/julia-feedstock/issues/24

Changes the recipe to make use of [`openspecfun`]( https://github.com/conda-forge/openspecfun-feedstock ). Also as that is the last Fortran dependency, drop `gcc` as the system toolchain should now be sufficient for building Julia. Trigger a rebuild of the package under build number 4 given these changes.

Also fixes the disconcerting `warning: parsing line table prologue at ... should have ended at ... but it ended at ...` issue that we have been seeing. Appears this was caused by using the `gcc` package. Using the system toolchain does not encounter the same issues.

Due to fixing the aforementioned warning and the removal of the Fortran dependency, this should help unblock 0.6.0.

### CODE REVIEW

File: recipe/build.sh [hunk 1]
```diff
@@ -6,10 +6,11 @@ export LIBRARY_PATH=${PREFIX}/lib
 export CMAKE_PREFIX_PATH=${PREFIX}
 
 make -j 4 prefix=${PREFIX} MARCH=core2 sysconfigdir=${PREFIX}/etc NO_GIT=1 \
+ LIBBLAS=-lopenblas LIBBLASNAME=libopenblas.so LIBLAPACK=-lopenblas LIBLAPACKNAME=libopenblas.so \
```
Comment (isuruf) [inline recipe/build.sh]: Are these really needed?

Comment (jakirkham) [reply recipe/build.sh]: I believe so. They had come from the `staged-recipes` PR ( https://github.com/conda-forge/staged-recipes/pull/3503 ), but I'm not finding the part in the conversation where these were added. In any event, just changing where they show up here not adding them outright.

**Comment by tkelman** [comment]
> the system toolchain should now be sufficient for building Julia

That's a bad idea and I'd advise against it. LLVM and Julia should be built with c++11 capable compilers.

**Comment by jakirkham** [comment]
We use C++11 supporting compilers on both macOS and Linux.

**Comment by tkelman** [comment]
Is it the devtoolset with its messed up handling of libstdc++? We don't recommend using that, especially not for creating anything you plan on redistributing

**Comment by jakirkham** [comment]
Are there reported issues that you have encountered with devtoolset and Julia or is this personal preference?

**Comment by tkelman** [comment]
The way it splits its libstdc++ caused problems. Was quite some time ago, would have to look up the exact issues.

**Comment by jakirkham** [comment]
FWIW tried grepping the Julia tracker and saw mention of such an issue, but didn't find the issue itself. Will keep an eye on it. Though I think right now we are in a better situation than we were having fixed the warnings that were being produced.

Some people at Continuum are working on newer compilers that we will be switching to. This work is nearing its completion FWIK. So I don't think we will be using devtoolset compilers for much longer.

**Pull request merged**